cogs/GlobalConfig.py

```diff
@@ -1,25 +1,64 @@
+import json
 class GlobalConfig():
-    def __init__(self, whitelist, trustedRoles, logChannelName, moderationChannelName, reportingChannelsList, OAuthToken, rolemenuData, lockedChannels):
-        self.whitelist = whitelist
-        self.trustedRoles = trustedRoles
-        self.logChannelName = logChannelName
-        self.moderationChannelName = moderationChannelName
-        self.reportingChannelsList = reportingChannelsList
-        self.OAuthToken = OAuthToken
-        self.rolemenuData = rolemenuData
-        self.lockedChannels = lockedChannels        
-
+    def __init__(self):
+        self.whitelist = []
+        self.trustedRoles = []
+        self.logChannelName = ""
+        self.moderationChannelName = ""
+        self.reportingChannelsList = []
+        self.reportingChannels = {}
+        self.OAuthToken = None
+        self.roleMenuData = {}
+        self.lockedChannels = []   
         self.reactions = "🇦 🇧 🇨 🇩 🇪 🇫 🇬 🇭 🇮 🇯 🇰 🇱 🇲 🇳 🇴 🇵 🇶 🇷 🇸 🇹 🇺 🇻 🇼 🇽 🇾 🇿".split()
-
+        self.permsError = "You don't have permission to use this command"
         # Source files cannot be removed and will not show up with a listfiles command, but they can be overwritten
         self.sourceFiles = [".git", ".gitignore", "config.json", "bot.py", "README.md", "Examples", "updatebot.sh", "LICENCE", "locked.dat", "rolemenu.dat"]
 
-        self.permsError = "You don't have permission to use this command"
+    # Parse all configs
+    def parseAll(self, configFilePath, roleMenuFilePath, lockedChannelFilePath):   
+        self._parseConfig(configFilePath)
+        self._parseRoleMenuData(roleMenuFilePath)
+        self._parseLockedChannelData(lockedChannelFilePath)
 
+    # Parse main config
+    def _parseConfig(self, filePath):
         # Prepare reporting channels
-        self.reportingChannels = {}
-        for i in self.reportingChannelsList:
-            self.reportingChannels[i[0]] = [i[1], i[2]]
+        def prepReportingChannels():
+            for i in self.reportingChannelsList:
+                self.reportingChannels[i[0]] = [i[1], i[2]]
+        try:
+            f = open(filePath)
+            data = json.load(f)
+            self.whitelist = data["whitelisted"]
+            self.trustedRoles = data["trustedRoles"]
+            self.logChannelName = data["logChannel"]
+            self.moderationChannelName = data["moderationChannel"]
+            self.reportingChannelsList = data["reportingChannels"]
+            self.OAuthToken = data["OAuth"]
+            prepReportingChannels()
+            f.close()
+        except Exception as e:
+            raise Exception(e)
+
+    # Parse role menu data
+    def _parseRoleMenuData(self, filePath):
+        try:
+            f = open(filePath)
+            self.rolemenuData = json.load(f)
+            f.close()
+        except Exception:
+            self.rolemenuData = {}
```

Comment:
```suggestion
            pass
```

If we've already set this as a default value in `__init__()` do we need to set it explicitly? This would probably be fine here right?

Reply:
That's fair that's fair. I can't think of a use case off the top of my head but I guess it's best to be explicit